### DISCUSSION TITLE
Use Hook Form - registering into the object

### DISCUSSION THREAD

**Discussion by tabrizh** [react-hook-form/react-hook-form#10080] [Q&A]
I am using react-hook-form with Typescript. The structure is complicated. I have to create the form dynamically from the api, i.e. all questions come from the api then I render the form based on that. So, I am using useFormContext. I fetch the data in the parent component, then I have another component to filter out the input types, and then based on that type I render either input or select and etc. After filtering out the input type, I send the whole Question object inside the Custom Input component where I render it. Questions have a value field that needs to be updated. The requirement is that I must send the same data I receive with only updated value field. 
Basically, I need to register the input to the Question Object's value field. I am getting this Question object as a prop, and when I try to register it this way: 
<Input
            key={question.question_id}
            type={`${question.isDateTime && "date"}`}
            defaultValue={question.value && question.value}
            {...register(`${question}.value` as const, {
              required: question.required,
              minLength: 2,
            })}
          />
it does not give any result.
Question looks like this: 
question_id: 1,
          question_title: "Last name",
          question_description: "",
          on_answer: null,
          questionKey: "lastName",
          required: false,
          widget_id: 5,
          widget: "input",
          value: "",
          answers: [],

**Comment by leapful** [comment]
@tabrizh 

Please provide a codesandbox url to re-produce the issue. Given just an incomplete code block does not help to investigate the problem.
